sphinx/texinputs/sphinxpackageboxes.sty

```diff
@@ -38,79 +80,358 @@
     \AtEndDocument{\PackageWarningNoLine{sphinx}{%
        I issued a warning which may have gotten lost in the\MessageBreak
        gigantic console output: pict2e.sty was not found,\MessageBreak
-       and verbatimradius has been ignored}}\endinput}
-
-% First we define some wrapper to be able to use arguments being (only)
-% dimensions or dimensional expressions.  The \unitlength will always be 1pt.
-\def\spx@moveto(#1,#2)%
-   {\moveto(\strip@pt\dimexpr#1,\strip@pt\dimexpr#2)}
-\def\spx@lineto(#1,#2)%
-   {\lineto(\strip@pt\dimexpr#1,\strip@pt\dimexpr#2)}
-% attention we use here [2] always; and there are two more mandatory
-% arguments, angles, we don't need to worry about them here.
-\def\spx@circlearc#1#2#3{\circlearc[2]%
-    {\strip@pt\dimexpr#1}{\strip@pt\dimexpr#2}{\strip@pt\dimexpr#3}%
+       and verbatimradius has been ignored}}%
+   \def\spx@boxes@fcolorbox@rounded{\spx@boxes@fcolorbox}%
+   }%
+}%
+
+
+%//// \spx@boxes@fcolorbox
+% This box will have the same baseline as its argument (which is typeset in
+% horizontal mode).  It takes into account four border widths parameters, four
+% padding parameters, two shadow widths (each possibly negative), and three
+% colors: background, border and shadow.  Its boundary box takes into account
+% all of shadow, border and padding.
+%
+% The customization of the various parameters are under responsability of
+% caller.  At some point of code development the parameters were set
+% by expansion of an \spx@boxes@fcolorbox@setup macro inside
+% \spx@boxes@fcolorbox.  Now this is only a hook, because the initialization
+% will be done by caller rather.
+\let\spx@boxes@fcolorbox@setuphook\@empty
+% The same applies to \spx@boxes@fcolorbox@rounded, also with (same name)
+% \spx@boxes@fcolorbox@setup left to caller configuration.
+%
+% The parameters are interpreted as they would as CSS properties.
+% For the shadow inset variant see separate \spx@boxes@fcolorbox@insetshadow
+\long\def\spx@boxes@fcolorbox#1{%
+  \hbox\bgroup
```

Comment:
this \hbox is started here and ended in another macro depending whether with inset shadow or not